fractal_server/app/runner/__init__.py

```diff
@@ -240,9 +171,16 @@ def _atomic_task_factory(
     task_executor = task.executor
     if task_executor is None:
         task_executor = settings.PARSL_DEFAULT_EXECUTOR
-    if task_executor not in valid_executors:
+    task_executor = add_prefix(
+        workflow_id=workflow_id, executor_label=task_executor
+    )
+
+    # Verify match between task_executor and available executors
+    valid_executor_labels = DataFlowKernelLoader.dfk().executors.keys()
+    if task_executor not in valid_executor_labels:
         raise ValueError(
```

Comment:
Non blocking: I think we should spin off everything that concerns setting up executors and parsl related stuff into its own routines and keep only a single call in `_atomic_task_factory`. As the name suggests, that function should only be concerned with preparing the tasks themselves, irrespectively of where they run.

in any case we can take care of separation of concerns in a subsequent refactoring.

Reply:
Fixed (?) with 09a5e798ab1a9de24a15c984119e254f08a24fa5. Now it looks like:
https://github.com/fractal-analytics-platform/fractal-server/blob/09a5e798ab1a9de24a15c984119e254f08a24fa5/fractal_server/app/runner/__init__.py#L149-L173